benchmarking/perception_benchmark_tool/perception_benchmark_tool/benchmark_tools/datasets_ros_nodes/waymo_ros_node.py

```diff
@@ -0,0 +1,458 @@
+            PointCloud2, "/point_cloud/side_right_lidar", 10
+        )
+        self.pub_lidar_top = self.create_publisher(PointCloud2, "/point_cloud/top_lidar", 10)
+
+        self.pub_gt_objects = self.create_publisher(DetectedObjects, "/gt_objects", 10)
+
+        self.point_fields = [
+            PointField(name="x", offset=0, datatype=PointField.FLOAT32, count=1),
+            PointField(name="y", offset=4, datatype=PointField.FLOAT32, count=1),
+            PointField(name="z", offset=8, datatype=PointField.FLOAT32, count=1),
+        ]
+
+        self.dataset = WaymoDataset(self.param_path)
+        self.pose_broadcaster = TransformBroadcaster(self)
+        self.static_tf_publisher = StaticTransformBroadcaster(self)
+        self.sub_tracking = self.create_subscription(
+            TrackedObjects,
+            "/perception/object_recognition/tracking/objects",
+            self.tracked_objects_callback,
+            10,
+        )
+
+        self.waymo_evaluation_frame = "base_link"
+
+        self.tf_buffer = Buffer()
+        self.tf_listener = TransformListener(self.tf_buffer, self)
+
+        self.prediction_proto_objects = metrics_pb2.Objects()
+        self.gt_proto_objects = metrics_pb2.Objects()
+
+    def tracked_objects_callback(self, tracked_objects):
+
+        for tracked_object in tracked_objects.objects:
+            if tracked_objects.header.frame_id != self.waymo_evaluation_frame:
+                tracked_object = self.transform_tracked_object(
+                    tracked_object, tracked_objects.header
+                )
+
+            tracked_object_waymo = metrics_pb2.Object()
+            tracked_object_waymo.context_name = self.context_waymo
+            tracked_object_waymo.frame_timestamp_micros = self.scene_timestamp
+            bbox_waymo = label_pb2.Label.Box()
+            bbox_waymo.center_x = tracked_object.kinematics.pose_with_covariance.pose.position.x
+            bbox_waymo.center_y = tracked_object.kinematics.pose_with_covariance.pose.position.y
+            bbox_waymo.center_z = tracked_object.kinematics.pose_with_covariance.pose.position.z
+            bbox_waymo.length = tracked_object.shape.dimensions.x
+            bbox_waymo.width = tracked_object.shape.dimensions.y
+            bbox_waymo.height = tracked_object.shape.dimensions.z
+
+            roll, pitch, yaw = euler_from_quaternion(
+                tracked_object.kinematics.pose_with_covariance.pose.orientation.x,
+                tracked_object.kinematics.pose_with_covariance.pose.orientation.y,
+                tracked_object.kinematics.pose_with_covariance.pose.orientation.z,
+                tracked_object.kinematics.pose_with_covariance.pose.orientation.w,
+            )
+
+            bbox_waymo.heading = yaw
+            tracked_object_waymo.object.box.CopyFrom(bbox_waymo)
+            tracked_object_waymo.score = 0.5
+            if tracked_object.classification[0].label == ObjectClassification.CAR:
```

Comment:
[IMO] How about making function like `convert_waymo2autoware_label()`. Currently, this is duplicated below and it will also help in the other evaluation task for waymo dataset.